python/pyspark/streaming/tests.py

```diff
@@ -61,9 +61,12 @@ def setUpClass(cls):
     def tearDownClass(cls):
         cls.sc.stop()
         # Clean up in the JVM just in case there has been some issues in Python API
-        jSparkContextOption = SparkContext._jvm.SparkContext.get()
-        if jSparkContextOption.nonEmpty():
-            jSparkContextOption.get().stop()
+        try:
```

Comment:
I can log the exception, it's just for the most part those references are
gone after the stop() function has been executed so it fails and prints out
a scary warning when the failure is expected instead and really it's more
concerning if it needed to be run (or at least this is my understanding
from poking at stuff last night).

On Tuesday, October 20, 2015, Josh Rosen <notifications@github.com> wrote:

> In python/pyspark/streaming/tests.py
> <https://github.com/apache/spark/pull/8615#discussion_r42521308>:
>
> > @@ -61,9 +61,12 @@ def setUpClass(cls):
> >      def tearDownClass(cls):
> >          cls.sc.stop()
> >          # Clean up in the JVM just in case there has been some issues in Python API
> > -        jSparkContextOption = SparkContext._jvm.SparkContext.get()
> > -        if jSparkContextOption.nonEmpty():
> > -            jSparkContextOption.get().stop()
> > +        try:
>
> Is this try block necessary? If so, would you mind logging the exception
> in the except branch, just so that it's not swallowed silently?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/8615/files#r42521308>.
>


-- 
Cell : 425-233-8271
Twitter: https://twitter.com/holdenkarau